### PR TITLE
Update spring-boot version to 2.5.8 and other dependencies

### PR DESCRIPTION
**Pull request by DanielFran**
Depends on https://github.com/jhipster/jhipster-bom/pull/587

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
@mraible @mshima Do you have a clue why is this failing?

**Comment by mraible** [comment]
@DanielFran I took a quick look and didn't see anything obvious.

**Comment by mshima** [comment]
@DanielFran have you tested locally?
Looks a bug at user/account for mysql/mariadb support.
Have you tried locally? I will try soon.

**Comment by mshima** [comment]
Looks like hibernate is not populating the user database.

File: generators/generator-constants.js [hunk 1]
```diff
@@ -36,14 +36,14 @@ const JIB_VERSION = '3.1.4';
 // Libraries version
 const JHIPSTER_DEPENDENCIES_VERSION = '7.4.2-SNAPSHOT';
 // The spring-boot version should match the one managed by https://mvnrepository.com/artifact/tech.jhipster/jhipster-dependencies/JHIPSTER_DEPENDENCIES_VERSION
-const SPRING_BOOT_VERSION = '2.5.7';
-const LIQUIBASE_VERSION = '4.6.1';
+const SPRING_BOOT_VERSION = '2.5.8';
+const LIQUIBASE_VERSION = '4.6.2';
```
Comment (mshima) [inline generators/generator-constants.js:40]:
```suggestion
const LIQUIBASE_VERSION = '4.6.1';
```

**Comment by DanielFran** [comment]
Thanks @mshima

**Comment by DanielFran** [comment]
Confirmed the issue was on liquibase side, there is an issue with compatibility with hibernate 2.6.2+ : https://github.com/liquibase/liquibase-hibernate/issues/378

**Pull request merged**